SCClassLibrary/External/Ableton/LinkClock.sc

```diff
@@ -0,0 +1,77 @@
+// clock with Ableton Link synchronization
+
+LinkClock : TempoClock {
+
+	var <>tempoChanged;
+
+	*newFromTempoClock { |clock|
+		^super.new(
+			clock.tempo,
+			clock.beats,
+			clock.seconds,
+			clock.queue.maxSize
+		).initFromTempoClock(clock)
+	}
+
+	initFromTempoClock { |clock|
+		var oldQueue;
+		//stop TempoClock and save its queue
+		clock.stop;
+		oldQueue = clock.queue.copy;
+		this.setMeterAtBeat(clock.beatsPerBar, 0);
+
+		forBy(1, oldQueue.size-1, 3) { |i|
+			var task=oldQueue[i+1];
+			//reschedule task with this clock
+			this.schedAbs(oldQueue[i], task);
+		};
+
+		^this
+	}
+
+
+	numPeers {
+		_LinkClock_NumPeers
+		^this.primitiveFailed
+	}
+
+	//override TempoClock primitives
+	beats_ { |beats|
+		_LinkClock_SetBeats
+		^this.primitiveFailed
+	}
+
+	setTempoAtBeat { |newTempo, beats|
+		_LinkClock_SetTempoAtBeat
+		^this.primitiveFailed
+	}
+	setTempoAtSec { |newTempo, secs|
+		_LinkClock_SetTempoAtTime
+		^this.primitiveFailed
+	}
+
+	setMeterAtBeat { |newBeatsPerBar, beats|
+		beatsPerBar = newBeatsPerBar;
+		barsPerBeat = beatsPerBar.reciprocal;
+		this.prSetQuantum(beatsPerBar);
+		this.changed(\meter);
+	}
+
+	// PRIVATE
+	prStart { |tempo, beats, seconds|
+		_LinkClock_New
+		^this.primitiveFailed
+	}
+
+	prSetQuantum { |quantum|
+		_LinkClock_SetQuantum;
+		^this.primitiveFailed
+	}
+
+	// run tempo changed callback
+	prTempoChanged { |tempo, beats, secs, clock|
+		tempoChanged.value(tempo, beats, secs, clock);
```

Comment:
wondering if this should be:

```suggestion
		tempoChanged.value(tempo, beats, secs, clock);
		this.changed(\tempo);
```

Other changes to the tempo in `TempoClock` send that notification. If `LinkClock` had, then my performance patch using `TempoClock` would have been able to be Link enabled by just changing the class of its clock object!

Reply:
Yes, `changed(\tempo)` should be mandatory.

Also, if the link quantum changes, then we need to update beatsPerBar and also do `this.changed(\meter)`.

The meter change could be tricky because of `baseBarBeats`.

Reply:
Fix proposed: https://github.com/jamshark70/supercollider/commit/a636e7b886d7ecc81f5121ef0239ed58094cd0f5

Also, Link (seemingly by design) does not propagate quantum changes. So we're off the hook for the meter-change question.